test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+PINECONE_ENVIRONMENT="gcp-starter"
+PINECONE_INDEX="ingest-test"
```

Comment:
does this mean every test running is writing to the same index? if so: this will very likely result in collisions when multiple run at the same time?

Reply:
Agreed, addressed with commit: [random index names to avoid test run collisions](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/35be64a36682c00b4b0b8b571043c69964290ea2)

Reply:
@ryannikolaidis , per the parallel index creation collision concerns:

### Multiple indexes are not supported in pinecone free tier

Even after we randomized our index names, we won't be able to create multiple indexes with the free tier:

1. One index per free (starter) project
> --requestCapacity Reached. Starter Projects support a single index. Create a new project to add more. Your Starter Project remains free post-upgrade.

2. One project per free org
![image](https://github.com/Unstructured-IO/unstructured/assets/39141206/699ba18e-8381-4bf0-9fea-6f724b5f0a60)

#### To be able to keep in the free tier (one index at a time)
- Keeping pinecone test for python 3.10 only (rather than all matrix) should help. 
- I also added an index creation retry logic so that index creation can be retried for up to 2 minutes when parallel tests are being triggered at the same time by different contributors. 
  - Pinecone test runs in 90 seconds in my machine and deletion of an index takes 10 to 15 seconds, totalling to <105 seconds. 120 seconds should provide enough leeway while still keeping the retry time as low as possible. It'd support:
    - up to 2 contributors triggering the CI at the exact same time,
    - unlimited contributors triggering CI, as long as it is with >105 second intervals

**Note**: Multiple index creation collisions are not something we've encountered in CI yet since pinecone test only runs in python 3.10. I assume we will encounter it when multiple contributors trigger CI at the same time, thus made the changes mentioned above.

Reply:
right, this is a separate concern, and sounds like we need to bump to a paid tier.
probably best not to burn CI time and people time for the sake of staying on the free tier.

> pinecone test only runs in python 3.10.
right, that and the fact that this hasn't merged, so this branch is literally the only branch test it..

Reply:
Updated to the paid tier now 👍